### PR TITLE
feat: add PR workflows to edx-arch-experiments

### PR DESCRIPTION
**Pull request by rgraber**
Add quality and commitlint checks to PRs

### CODE REVIEW

File: edx_arch_experiments/kafka_consumer/events.py [hunk 1]
```diff
@@ -12,6 +12,10 @@
 # Use TrackingEvent class from openedx_events and use Attr <-> Avro bridge to deserialize
 class TrackingEvent:
 
+    """
+        Data class for license-manager subscription events
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/events.py:16]:
```suggestion
    Data class for license-manager subscription events
```

File: tox.ini [hunk 2]
```diff
@@ -31,15 +31,29 @@ match-dir = (?!migrations)
 
 [pytest]
 DJANGO_SETTINGS_MODULE = test_settings
-addopts = --cov edx_arch_experiments --cov-report term-missing --cov-report xml
-norecursedirs = .* docs requirements site-packages
+addopts = --cov edx_braze --cov-report term-missing --cov-report xml
```
Comment (timmc-edx) [inline tox.ini:34]: Unintentional copy/paste?

File: edx_arch_experiments/kafka_consumer/events.py (hunk 1 above)
Comment (timmc-edx) [reply edx_arch_experiments/kafka_consumer/events.py:16]: (Same for some other docstrings in this PR.)

File: tox.ini (hunk 2 above)
Comment (rgraber) [reply tox.ini:34]: yes ><. good catch

File: edx_arch_experiments/kafka_consumer/settings/common.py [hunk 3]
```diff
@@ -1,6 +1,10 @@
-def plugin_settings(settings):
-    """Default settings values."""
+"""Default settings values."""
+
 
+def plugin_settings(settings):
+    """
+        Get kafka_consumer settings from calling application
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/settings/common.py:6]:
```suggestion
    Get kafka_consumer settings from calling application
```

File: edx_arch_experiments/kafka_consumer/management/commands/tests/test_consume_events.py [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 from django.test import TestCase
 from django.test.utils import override_settings
-from unittest.mock import patch
+
 from edx_arch_experiments.kafka_consumer.management.commands.consume_events import Command
 
 
 class TestConsumeEventCommand(TestCase):
+    """
+        Tests for the consume_events management command
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/management/commands/tests/test_consume_events.py:14]:
```suggestion
    Tests for the consume_events management command
```

File: edx_arch_experiments/kafka_consumer/management/commands/consume_events.py [hunk 5]
```diff
@@ -99,11 +104,17 @@ def create_consumer(self, group_id):
...
+        """
+            Place holder methods for how to handle an incoming message from the event bus
+        """
         # TODO (EventBus):
         # Rewrite this to construct and/or emit the signal eventually specified in the message.
         logger.info(f"Received message with key {msg.key()} and value {TrackingEvent.to_dict(msg.value())}")
 
     def process_single_message(self, msg):
+        """
+            Handle message error or pass along for processing (separated out for easier testing)
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/management/commands/consume_events.py:116]:
```suggestion
        Handle message error or pass along for processing (separated out for easier testing)
```

File: edx_arch_experiments/kafka_consumer/management/commands/consume_events.py [hunk 6]
```diff
@@ -99,11 +104,17 @@ def create_consumer(self, group_id):
         return DeserializingConsumer(consumer_config)
 
     def handle_message(self, msg):
+        """
+            Place holder methods for how to handle an incoming message from the event bus
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/management/commands/consume_events.py:108]:
```suggestion
        Place holder methods for how to handle an incoming message from the event bus
```

File: edx_arch_experiments/kafka_consumer/management/commands/consume_events.py [hunk 7]
```diff
@@ -64,6 +64,11 @@ def add_arguments(self, parser):
         )
 
     def create_consumer(self, group_id):
+        """
+            Create a consumer for TrackingEvents
+            :param group_id: id of the consumer group this consumer will be part of
+            :return: DeserializingConsumer
```
Comment (timmc-edx) [inline edx_arch_experiments/kafka_consumer/management/commands/consume_events.py:70]:
```suggestion
        Create a consumer for TrackingEvents
        :param group_id: id of the consumer group this consumer will be part of
        :return: DeserializingConsumer
```

**Review by timmc-edx — APPROVED** [review]

**Pull request merged**